src/panel/PanelManager.jsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SearchInput from '../ui_components/search_input';
+import nconf from '@qwant/nconf-getter';
+import FavoritesPanel from './favorites/FavoritesPanel';
+import PoiPanel from './PoiPanel';
+import ServicePanel from './ServicePanel';
+import EventListPanel from './event/EventListPanel';
+import CategoryPanel from 'src/panel/category/CategoryPanel';
+import DirectionPanel from 'src/panel/direction/DirectionPanel';
+import classnames from 'classnames';
+import { parseQueryString, getCurrentUrl } from 'src/libs/url_utils';
+
+const performanceEnabled = nconf.get().performance.enabled;
+const directionEnabled = nconf.get().direction.enabled;
+const categoryEnabled = nconf.get().category.enabled;
+const eventEnabled = nconf.get().events.enabled;
+
+export default class PanelManager extends React.Component {
+  static propTypes = {
+    router: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isMinified: false,
+      ActivePanel: ServicePanel,
+      options: {},
+    };
+  }
+
+  componentDidMount() {
+    this.initRouter();
+    this.initTopBar();
+    if (performanceEnabled) {
+      window.times.appRendered = Date.now();
+    }
+  }
+
+  componentDidUpdate(_prevProps, prevState) {
```

Comment:
This code is still hard to read as it includes our very specific rules about panel switching: auto minification of the search bar and the damned stuff about persistent category markers…